src/test/java/org/opensearch/sdk/sample/helloworld/TestHelloWorldExtension.java

```diff
@@ -46,4 +75,34 @@ public void testExtensionRestHandlers() {
         assertEquals(2, routes.size());
     }
 
+    /**
+    @Test
+    public void testCreateComponents() throws IOException {
+        Settings settings = Settings.builder().put("node.name", Property.NodeScope).build();
+        ThreadPool threadPool = new ThreadPool(settings, new AtomicReference<>(), Collections.emptyList().toArray(new ExecutorBuilder[0]));
+        Client client = new NodeClient(settings, threadPool);
+        ClusterService clusterService = new ClusterService(settings, new ClusterSettings(settings, new HashSet<Setting<?>>()), threadPool);
+        ResourceWatcherService resourceWatcherService = new ResourceWatcherService(settings, threadPool);
+        ScriptService scriptService = new ScriptService(settings, new HashMap<String, ScriptEngine>(), new HashMap<String, ScriptContext<?>>());
+        NamedXContentRegistry xContentRegistry = new NamedXContentRegistry(new ArrayList<NamedXContentRegistry.Entry>());
+        Environment environment = new Environment(settings, Paths.get(""));
+        NodeEnvironment nodeEnvironment = new NodeEnvironment(settings, environment);
+        NamedWriteableRegistry namedWriteableRegistry = new NamedWriteableRegistryAPI().getRegistry();
+        IndexNameExpressionResolver indexNameExpressionResolver = new IndexNameExpressionResolver(threadPool.getThreadContext());
+        final SetOnce<RepositoriesService> repositoriesServiceReference = new SetOnce<>();
+        Supplier<RepositoriesService> repositoriesServiceSupplier = repositoriesServiceReference::get;
+        assertEquals(threadPool, extension.getThreadPool());
```

Comment:
I assume some instantiation will eventually live here.  A line break between the two sections would help readability.